spyder/plugins/switcher/widgets/switcher.py

```diff
@@ -293,14 +317,21 @@ def setup(self):
 
         # Filter by text
         titles = []
-        for row in range(self.model.rowCount()):
+        items_data = []
+        for row in range(self.model.rowCount()-1, -1, -1):
+            # As we are removing items from the model, we need to iterate
+            # backwards so that the indexes are not affected
             item = self.model.item(row)
             if isinstance(item, SwitcherItem):
-                title = item.get_title()
+                items_data.append(item._data._filename.lower())
+                if item._section == "Projects":
+                    self.model.removeRow(row)
+                    continue
+                else:
+                    title = item.get_title()
```

Comment:
No, I made a mistake when adding this line `items_data.append(item._data._filename.lower())`, because items_data should be filled only with open files in editor (which means everything, but Projects section). So this part of the code would be like this: 
```py
if item._section == "Projects":
    self.model.removeRow(row)
    continue
else:
    title = item.get_title()
    items_data.append(item._data._filename.lower())
```